test/test_autograd.py

```diff
@@ -4282,6 +4282,76 @@ def manual_increase_gradient(grad):
         loss.backward()
         self.assertEqual(param.grad, 6 * param)
 
+    def test_grad_fn_attr_bindings(self):
+        # Check that the getter of each type returns what we want
+        # See `gen_autograd_functions.py` for how getters are generated
+        a = torch.ones(1, requires_grad=True)
+        b = torch.ones(1, requires_grad=True)
+        out = torch.stack([a, b], dim=0)
+        self.assertEqual(out.grad_fn._tensors, (a, b))              # TensorList -> Tuple[Tensor]
+        self.assertIsInstance(out.grad_fn._tensors, tuple)
```

Comment:
The assertEqual above should already be checking for the type and content no?
Same for al lthe tests, you can remove the duplicate checks (if any)

Reply:
Removed the tuple ones. the assertEqual doesn't check types are the same for numeric types though